Simplenote/NoticeView.swift

```diff
@@ -0,0 +1,93 @@
+import UIKit
+
+protocol NoticePresentingDelegate: class {
+    func noticePressBegan()
+    func noticePressEnded()
+}
+
+class NoticeView: UIView {
+
+    // MARK: Properties
+    //
+    @IBOutlet weak var stackView: UIStackView!
+    @IBOutlet weak var noticeLabel: UILabel!
+    @IBOutlet weak var noticeButton: UIButton!
+
+    weak var delegate: NoticePresentingDelegate?
+    var action: (() -> Void)? {
+        didSet {
+            noticeButton.isHidden = action == nil
+        }
+    }
+
+    // MARK: Initialization
+    //
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        setupView()
+    }
+
+    // MARK: View Layout
+    //
+    private func setupView() {
+        translatesAutoresizingMaskIntoConstraints = false
+        setupViewStyles()
+        setupLongPress()
+
+        layoutIfNeeded()
+    }
+
+    private func setupLongPress() {
+        let longPressGesture = UILongPressGestureRecognizer(target: self, action: #selector(viewWasLongPressed(_:)))
+        addGestureRecognizer(longPressGesture)
+    }
+
+    private func setupViewStyles() {
+        stackView.backgroundColor = .simplenoteNoticeViewBackgroundColor
+        stackView.layer.cornerRadius = Constants.cornerRadius
+        stackView.clipsToBounds = true
+
+        noticeLabel.textColor = .simplenoteTextColor
+        noticeButton.setTitleColor(.simplenoteTintColor, for: .normal)
+        noticeButton.setTitleColor(.simplenoteCardDismissButtonHighlightedBackgroundColor, for: .highlighted)
+        noticeButton.isHidden = true
+
+    }
+
+    // MARK: Action
+    //
+    @IBAction func noticeButtonWasTapped(_ sender: Any) {
+        action?()
+    }
+}
+
+// NOTE: long press recognizing has not been connected to anything yet
+// Currently just prints to log that a press event happened.
```

Comment:
Just curious, what will be the purpose of long press?

Reply:
It interrupts the automatic dismissal of the notice.  So if the user wants the notice to not disappear yet, as long as they press on the notice it will stay

Reply:
The notice is not very big, so I wonder if you press it with finger, can you still read it? 🤔